### PR TITLE
feat: allow uploading on main channel for v1 recipes

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

now that we have implemented all rattler-build lints checks and we have a working feedstock: https://github.com/conda-forge/jolt-physics-feedstock
I would like to uplift restrictions of uploading rattler-build to main channel.

Should be merged after release of: https://github.com/conda-forge/conda-smithy/pull/2037

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
We need to wait a bit imho for the dust to settle on the linter. There are a few PRs left. Also the bot is not yet finished, which might break things.

**Comment by nichmor** [comment]
> We need to wait a bit imho for the dust to settle on the linter. There are a few PRs left. Also the bot is not yet finished, which might break things.

Hey ! Thanks for the review!


Can we make an actionable plan for linter work? 
I know that there is https://github.com/conda-forge/conda-smithy/pull/2028 work that needs to be done.
Do you have anything else on your radar?

Regarding the bot work- can it block us from allowing us to upload on the main channel? I think that even we break things we do it on bot side, uploaded channels will work just fine?

**Comment by beckermr** [comment]
It depends on how the bot breaks. If people convert feedstocks that need ABI migrations or version migrations, we'll break more than just a single feedstock. Things that affect the whole ecosystem like this have to move slowly enough for folks to adapt.

**Comment by baszalmstra** [comment]
@beckermr I completely understand where you are coming from but I feel the issue of packages not being uploaded is a bit of a chicken-and-egg problem. We have already had people (in the chat) wanting to convert their feedstock but holding off because the packages would not get uploaded. That makes it hard for the dust to settle. I know @jaimergp also had different ideas about this.

From my perspective, these are the issues that are left open:

conda-smithy:

- [ ] https://github.com/conda-forge/conda-smithy/pull/2037
- [ ] https://github.com/conda-forge/conda-smithy/pull/2028
- [ ] https://github.com/conda-forge/conda-smithy/pull/2036

bot:

- [ ] https://github.com/regro/cf-scripts/pull/2920

Am I missing issues or pull-requests?

**Comment by beckermr** [comment]
The ABI migrations run mini migrators on top of the recipes. We'll need to merge the prs to skip those or make them work on the new format.

**Comment by baszalmstra** [comment]
> The ABI migrations run mini migrators on top of the recipes. We'll need to merge the prs to skip those or make them work on the new format.

Is there already an issue for this so we can keep track of it?

**Comment by beckermr** [comment]
On the bottom repo it is issues 2918 2920 and 2924

**Comment by baszalmstra** [comment]
Thanks I updated the overview. 👍

**Comment by beckermr** [comment]
Another issue is that the solver checks repo does not currently work for v1 recipes: https://github.com/regro/conda-forge-feedstock-check-solvable/issues/54

Do we have a tracking issue on the website repo for this? If not we should.

**Comment by wolfv** [comment]
Some smaller progress was made wrt to the CODEOWNERS file - a change in rattler-build-conda-compat _should_ have made it work.

**Comment by wolfv** [comment]
I bumped the version to `4.9.5` - would love to see this merged :)

File: recipe/meta.yaml
Comment (beckermr) [inline recipe/meta.yaml:2]:
```suggestion
{% set version = "4.10.0" %}
```

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
@conda-forge/core shall we go ahead with this? I am a +1.

**Review by h-vetinari — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]
Am in favor of trying this

Would suggest we do this on a non-weekend adjacent day (so not Monday or Friday). For example we could do this tomorrow (Thursday morning) or some time next week (Tuesday or later). Mention this as most people likely don't want to spend their weekends debugging and fixing issues. So picking some time where people will be around and ready to work on issues would be best

**Comment by beckermr** [comment]
I'm around for the next four hours so I'll merge this now and watch things on the heroku server.

**Comment by jakirkham** [comment]
Sounds good. Thanks Matt and thanks everyone working on this upgrade! 🙏

**Comment by jakirkham** [comment]
Noticed the `conda-forge-ci-setup` cross-compiled architectures are failing (for [example]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037636&view=logs&jobId=bbe49da5-7bd7-508d-ebc8-759d9757842e&j=bbe49da5-7bd7-508d-ebc8-759d9757842e&t=6c37600a-1f14-5752-4b6b-025cf34b55df ))

Do we need to bootstrap this change somehow (wait for native builds and restart)? Or is there something missing in those cases?

**Comment by beckermr** [comment]
That appears to be a cross-python issue unrelated to this package, right?

cc @conda-forge/core @conda-forge/cross-python

**Comment by jakirkham** [comment]
Yeah came to the same conclusion

That package has this logic in [its activation script]( https://github.com/conda-forge/cross-python-feedstock/blob/d94169b29f29de1b24dc39576ac675dac8a6d048/recipe/activate-cross-python.sh#L32-L38 ):

```bash
    $BUILD_PREFIX/bin/python -m crossenv $PREFIX/bin/python \
        --sysroot $CONDA_BUILD_SYSROOT \
        --without-pip $BUILD_PREFIX/venv \
        --sysconfigdata-file "$sysconfigdata_fn" \
	--machine ${machine} \
        --cc ${CC:-@CC@} \
        --cxx ${CXX:-@CXX@}
```

Am guessing `$CONDA_BUILD_SYSROOT` is undefined when building with `rattler-build`. Though maybe there is another change or explanation that I'm missing

**Comment by beckermr** [comment]
We don't build with rattler build in ci-setup, so it is something else.

**Comment by jakirkham** [comment]
AFAIK we don't set `CONDA_BUILD_SYSROOT` here (with the exception of macOS)

Note this error happens for the CPU-only ARM/PPC builds here. In case that provides any clues

Separately: Maybe we should switch `conda-forge-ci-setup` to use `rattler-build`? It would be good to have a check here to confirm that we are able to build with `rattler-build` before publishing new `conda-forge-ci-setup` packages

**Comment by beckermr** [comment]
I think this is conda-build 24.9

**Comment by beckermr** [comment]
Something shipped today broke everything. The CI ran fine this morning.

**Comment by jakirkham** [comment]
Could this be related ( https://github.com/conda-forge/binutils-feedstock/issues/76 )?

**Comment by jakirkham** [comment]
Discussion continues in PR: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/349

**Comment by jakirkham** [comment]
In any event, think we agree this is not related to the changes in this PR. Just external changes that are now affecting `conda-forge-ci-setup`

Given that, decided to kick the tires on this change. Recently, while I was out, my colleagues added a brand new feedstock [`numba-cuda` to conda-forge]( https://github.com/conda-forge/numba-cuda-feedstock ), which uses the new recipe format. Of course no packages built from that at the time. Started [a fresh CI build]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037705&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3 ) to see if it would upload packages. It appears to [have worked]( https://anaconda.org/conda-forge/numba-cuda/files )! 🎉 Will work with them on testing these packages out

Also have started [a fresh build of `jolt-physics`]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037707&view=results ). Let's keep an eye on how that goes

**Comment by jakirkham** [comment]
> Also have started [a fresh build of `jolt-physics`](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037707&view=results). Let's keep an eye on how that goes

This unfortunately [failed]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1037707&view=results ). Though for reasons that appear to have to do with that feedstock itself

Am guessing it just needs PR ( https://github.com/conda-forge/jolt-physics-feedstock/pull/2 ) to fix it

**Pull request merged**
